src/Timer.cc

```diff
@@ -56,11 +56,45 @@ Time Timer::Elapsed() const
 {
   if (this->running)
   {
-    Time currentTime;
-    currentTime = Time::SystemTime();
+    std::chrono::steady_clock::time_point currentTime;
+    currentTime = std::chrono::steady_clock::now();
+    std::chrono::duration<double> diff = currentTime - this->start;
+#ifndef _WIN32
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+    return Time(diff.count());
+#ifndef _WIN32
+# pragma GCC diagnostic pop
+#endif
+  }
+  else
+  {
+    std::chrono::duration<double> diff = this->stop - this->start;
+#ifndef _WIN32
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+    return Time(diff.count());
+#ifndef _WIN32
+# pragma GCC diagnostic pop
+#endif
+  }
+}
 
-    return currentTime - this->start;
+//////////////////////////////////////////////////
```

Comment:
Same here about avoiding duplication by having the old function call the new one. It will also reduce the number of pragmas.